app/lib/phdc/serializer.rb

```diff
@@ -0,0 +1,431 @@
+      end
+
+      # Ethnicity
+      unless patient.ethnicity.blank?
+        eth_code = patient.ethnicity.include?('Not') ? '2186-5' : '2135-2'
+        patient_details << code_helper(eth_code, r_oid, patient.ethnicity, 'Race & Ethnicity - CDC', :ethnicGroupCode)
+      end
+
+      # Author
+      author = Ox::Element.new(:author)
+      cd_root << author
+      auth_time = Ox::Element.new(:time)
+      auth_time['value'] = patient.created_at&.strftime('%Y%m%d%H%M%S%z')
+      author << auth_time
+      assigned_author = Ox::Element.new(:assignedAuthor)
+      author << assigned_author
+      auth_id = Ox::Element.new(:id)
+      auth_id['root'] = '2.16.840.1.113883.19.5'
+      assigned_author << auth_id
+      assigned_person = Ox::Element.new(:assignedPerson)
+      assigned_author << assigned_person
+      auth_name = Ox::Element.new(:name)
+      assigned_person << auth_name
+      jur_name = Ox::Element.new(:family)
+      jur_name << "Sara Alert NBS Export: #{jurisdiction.jurisdiction_path_string}"
+      auth_name << jur_name
+
+      # Custodian
+      custodian = Ox::Element.new(:custodian)
+      cd_root << custodian
+      assigned_custodian = Ox::Element.new(:assignedCustodian)
+      custodian << assigned_custodian
+      represented_custodian_organization = Ox::Element.new(:representedCustodianOrganization)
+      assigned_custodian << represented_custodian_organization
+      custodian_id = Ox::Element.new(:id)
+      custodian_id['root'] = '1.3.3.3.333.23'
+      represented_custodian_organization << custodian_id
+      jur_name_cust = Ox::Element.new(:name)
+      jur_name_cust << "Sara Alert NBS Export: #{jurisdiction.jurisdiction_path_string}"
+      represented_custodian_organization << jur_name_cust
+
+      # Body
+      outer_component = Ox::Element.new(:component)
+      cd_root << outer_component
+      structured_body = Ox::Element.new(:structuredBody)
+      outer_component << structured_body
+
+      # Social History Information Section
+
+      sh_component = Ox::Element.new(:component)
+      structured_body << sh_component
+      sh_section = Ox::Element.new(:section)
+      sh_component << sh_section
+      sh_id = Ox::Element.new(:id)
+      sh_id['root'] = '2.16.840.1.113883.19'
+      sh_id['extension'] = patient.id
+      sh_section << sh_id
+      sh_section << code_helper('29762-2', '2.16.840.1.113883.6.1', 'Social history', 'LOINC')
+      sh_section_title = Ox::Element.new(:title)
+      sh_section_title << 'SOCIAL HISTORY INFORMATION'
+      sh_section << sh_section_title
+      if patient.primary_language == 'English'
+        sh_section << entry_helper_code('DEM142', 'Patient Primary Language', 'CE', 'ENG', 'English')
+        sh_section << entry_helper_code('NBS214', 'Patient Speaks English', 'CE', 'Y', 'Yes')
+      end
+      unless patient.age.blank?
+        sh_section << entry_helper_text('INV2001', 'Patient Age Reported', 'ST', patient.age.to_s)
```

Comment:
We should probably use the calculated age here.

Reply:
Good idea, made this change.